lib/logstash/outputs/s3.rb

```diff
@@ -106,6 +109,13 @@ class LogStash::Outputs::S3 < LogStash::Outputs::Base
   # S3 bucket
   config :bucket, :validate => :string, :required => true
 
+  # Specify a custom endpoint for use with non-AWS S3 implementations, e.g.,
+  # Ceph.  Provide a URL in the format http://127.0.0.1:8080/
+  config :endpoint, :validate => :string
```

Comment:
Is "endpoint" the best name?

Is `http` the right scheme for the url? Citing existing exsample:

* `s3cmd` allows `s3://` uri
* `s3cmd` config has `host_base` and `host_bucket` for setting hostnames different from AWS S3.
* hadoop uses `s3a://` (and older `s3n` and `s3`)
* [boto](http://docs.ceph.com/docs/master/radosgw/s3/python/) allows users to set the `host` for the endpoint, but not a URL (based on my probably-incomplete research)

Reply:
Some users want HTTP on trusted networks and others HTTPS on the wider Internet.